exoscale/resource_exoscale_compute.go

```diff
@@ -225,50 +235,55 @@ func resourceComputeCreate(d *schema.ResourceData, meta interface{}) error {
 		return err
 	}
 
-	diskSize := int64(d.Get("disk_size").(int))
-	resp, err = client.RequestWithContext(ctx, &egoscale.ListTemplates{
-		TemplateFilter: "featured",
-		ZoneID:         zone.ID,
-	})
-	if err != nil {
-		return err
-	}
-
-	var templateID *egoscale.UUID
 	username := ""
-	currentDiskSize := diskSize << 30 // Gib to B
-	image := strings.ToLower(d.Get("template").(string))
-
-	// First try to parse the image value as a UUID, if it fails try as a name
-	if templateID, err = egoscale.ParseUUID(image); err != nil {
-		for _, template := range resp.(*egoscale.ListTemplatesResponse).Template {
-			// Skip non-machine images
-			if strings.ToLower(template.Name) != image {
-				continue
-			}
+	templateID := ""
+	_, byLegacyName := d.GetOk("template")
+	_, byID := d.GetOk("template_id")
+	if !byLegacyName && !byID {
+		return errors.New("either template or template_id must be specified")
+	}
+
+	// If template is specified by name, fetch the template details from the API to get its ID.
+	// This behavior relates to the deprecated `template` attribute and will be removed in the future,
+	// where the resource will only support template IDs looked up by the compute_template data source.
+	if byLegacyName {
+		resp, err = client.RequestWithContext(ctx, &egoscale.ListTemplates{
```

Comment:
For the Legacy way you can use the egoscale `GetWithContext`, that can reduce the code in this `if` statement

Reply:
Thanks, fixed :)